### ISSUE TITLE
Checkout filepath

### ISSUE THREAD

**Issue by jcubic** [isomorphic-git/isomorphic-git#201]
It seems that there is only option to checkout branch but not files.

**Comment by wmhilton** [comment]
Yep, we should do that! It would look like:

```js
let repo = {fs, dir: '.'}
await git.checkout({...repo, ref: 'master', filepath: 'oops/deleted/this.txt' })
```

But `ref` would be optional now because it could default to `HEAD`.

**Comment by joaopppobr** [comment]
Hi! I would like to address this, but I'm not familiar with the way to do it (first commit), could you guys give me some help?

**Comment by jcubic** [comment]
@joaopppobr take a look at src/commands/checkout.js it have function `writeTreeToDisk` you need to write similar function (it also need to traverse) but that will write only one file. You then call that funcction in checkout when filepath is set and don't call `await fs.write(`${gitdir}/HEAD`, `ref: refs/heads/${ref}`)`.

**Comment by jcubic** [comment]
You probably be better in writing traverse function with callback for each object to not duplicate code.

**Comment by erdahuja** [comment]
@wmhilton can you explain it using this example
```
{ filepath: '/var/folders/vs/qwfpw2tx28n3573_580tc49h00
00gn/T/jest-fixture-LHcYlv/a.txt',
        dir: '/var/folders/vs/qwfpw2tx28n3573_580tc49h0000gn/T/jest-fixture-LHcYlv' }

```

- from writeTreeToDisk();

and how to proceed with function. It's my first time open source.

**Comment by nelsonni** [comment]
It appears that function `writeTreeToDisk` was removed in #524 and the logic migrated inline into the `walkBeta1` traverse block. What is the preferred structure for adding the `filepath` option now, since re-adding the `writeTreeToDisk` function seems to be moving backwards?

**Comment by wmhilton** [comment]
It might be even easier now... we could just use the `filter` argument to `walkBeta1` similarly to how we do in [`statusMatrix`](https://github.com/isomorphic-git/isomorphic-git/issues/201). Then it would only check out files matching `pattern`!

**Comment by nelsonni** [comment]
PR #717 should correctly add the `pattern` parameter to the `checkout` command. Unit test is included.

**Comment by wmhilton** [comment]
Closed because of awesome PR #717
